external/mdns/mdns.c

```diff
@@ -242,19 +250,32 @@ static uint8_t *uncompress_nlabel(uint8_t *pkt_buf, size_t pkt_len, size_t off)
 	uint8_t *e = pkt_buf + pkt_len;
 	size_t len = 0;
 	char *str, *sp;
+	uint16_t cycle = 0;
 	if (off >= pkt_len) {
 		return NULL;
 	}
 	// calculate length of uncompressed label
 	for (p = pkt_buf + off; *p && p < e; p++) {
 		size_t llen = 0;
+		if (cycle++ >= DETECT_CYCLE) {
+			DEBUG_PRINTF("JB patched : uncompress_nlabel : DoS (loop in compressed labels)\n"); break;
```

Comment:
is there no need for immediate return with NULL?

Reply:
that would be better. I will fix